src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +438,124 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 
+{
+    if (String.IsNullOrWhiteSpace(stringValue)) {
+        return;
+    }
+
+    this.CustomHttpHeaders = new Dictionary<string, string>();
+
+    if (ValidateParantheses(stringValue) && ValidateFirstLast(stringValue) && ValidateDelimeters(stringValue)) 
+    {
+        string header = stringValue.TrimEnd('}', ' ');
+        string[] elements = header.Split('}');
+        foreach (var element in elements)
+        {
+            string _element = element.TrimStart(',', '{', ' ');
+            string[] array = _element.Split(':');
+            CustomHttpHeaders.Add(array[0], array[1]);
+        }
+    }
+    else 
+    {
+        throw new ArgumentException("Custom http headers string format should be '{headerName:value},{headerName:value}'");
+    }
+
+}
+
+private static bool ValidateParantheses(string _input)

```

Comment:
>string _input [](start = 40, length = 13)

It's not conventional to name a parameter name with "underscore".

Reply:
Fixed this